rpcs3/Emu/Cell/lv2/sys_rsx.cpp

```diff
@@ -437,13 +468,33 @@ error_code sys_rsx_context_attribute(u32 context_id, u32 package_id, u64 a3, u64
 	{
 	case 0x001: // FIFO
 	{
-		rsx::eng_lock rlock(render);
 		const u64 get = static_cast<u32>(a3);
 		const u64 put = static_cast<u32>(a4);
-		vm::_ref<atomic_be_t<u64>>(render->dma_address + ::offset32(&RsxDmaControl::put)).release(put << 32 | get);
-		render->fifo_ctrl->set_get(static_cast<u32>(get));
-		render->last_known_code_start = get;
-		render->sync_point_request.release(true);
+		const u64 get_put = put << 32 | get;
```

Comment:
What's going on here exactly? You pause FIFO then switch the dma pointers an arbitrary amount of time later, but no explanation is given. I don't see how this is safer than the previous approach.
The eng_lock will not return until FIFO is outside of a command sequence anyway.

Reply:
There is a slight issue with value reading when multiple threads write to it, will fix in a bit.

Reply:
Fixed.